src/algokit_utils/application_client.py

```diff
@@ -883,21 +881,9 @@ def _simulate_readonly_call(
         return TransactionResponse.from_atr(simulate_response)
 
     def _simulate_atc(self, atc: AtomicTransactionComposer) -> SimulateAtomicTransactionResponse:
```

Comment:
You should be able to remove this entire function, and call `atc.simulate(self.algo_client)` instead of `self._simulate_atc(atc)`